webapp/utils/__init__.py

```diff
@@ -52,14 +52,15 @@ def get_task_status_and_logs(task_id: str) -> tuple[str, list]:
             else status
         )
     except ECSTaskDoesNotExistError:
-        task_status = "COMPLETED"
+        task_status = "UNKNOWN"
 
-    # If logs cannot be found, task did not exist
-    logs = ["Waiting for logs."]
-    if task_status == "COMPLETED":
+    logs = ["Loading."]
+    if task_status in ["COMPLETED", "UNKNOWN"]:
         logs = cloudwatchlogs_client.get_log_messages(task_id)
+        if not logs:
+            return "EXPIRED (UNKNOWN)", ["Log stream expired, cannot find logs for task."]
 
-    return task_status, logs
+    return "COMPLETED", logs
```

Comment:
Wondering if I'm missing something here?  How could the status be anything _but_ `COMPLETE`, `UNKNOWN`, or `EXPIRED (UNKNOWN)`?

It seems as though any other `task_status` will be overridden by this line that sets it to `COMPLETE`, as this is the only part of the method that returns a value.

Here is the full method from the PR branch:

```python
def get_task_status_and_logs(task_id: str) -> tuple[str, list]:
  # docstring here...
  ecs_client = ECSClient()
  cloudwatchlogs_client = CloudWatchLogsClient()
  # If task exists, get the current status
  try:
      task_status = (
          "COMPLETED"
          if (status := ecs_client.get_task_status(task_id)) == "STOPPED"
          else status
      )
  except ECSTaskDoesNotExistError:
      task_status = "UNKNOWN"
  
  logs = ["Loading."]
  if task_status in ["COMPLETED", "UNKNOWN"]:
      logs = cloudwatchlogs_client.get_log_messages(task_id)
      if not logs:
          return "EXPIRED (UNKNOWN)", ["Log stream expired, cannot find logs for task."]
  
  return "COMPLETED", logs
```

I ran a "review" run in the deployed version, and ended up with this state, which doesn't feel correct:

<img width="276" alt="Screenshot 2024-09-04 at 9 36 34 AM" src="https://github.com/user-attachments/assets/34fc7a35-da63-4002-a09a-f01ca4b25459">

Reply:
Ah, I don't have a good reason for this error other than it being a misstep in an attempt to fix a recurring call to the "data" route. 😅 One of the updates I made resulted in a never-endling loop that sent a request to the "data" route -- even after the logs had been retrieved. I've fixed the bug via https://github.com/MITLibraries/alma-sapinvoices-ui/pull/83/commits/8e3743744c9438395f0001314bc199dbfc233bf3 and re-deployed to Dev1.